### PR TITLE
feat: tidb cloud serverless tier support

### PR DESCRIPTION
**Pull request by Icemap**
[TiDB](https://github.com/pingcap/tidb) (/’taɪdiːbi:/, "Ti" stands for Titanium) is an open-source distributed SQL database that supports Hybrid Transactional and Analytical Processing (HTAP) workloads. It is MySQL compatible and features horizontal scalability, strong consistency, and high availability.

I want to add a support module here. PTAL.
If you want to test it, [TiDB Cloud](https://tidbcloud.com/) Serverless Tier is free now. You can try it for totally free.

image
image
image

### CODE REVIEW

**Comment by Icemap** [comment]
I already tested it. Looks good. https://sqlchat-git-main-pingcap.vercel.app/

File: src/pages/api/connection/db.ts [hunk 1]
```diff
@@ -10,7 +11,7 @@ const handler = async (req: NextApiRequest, res: NextApiResponse) => {
     return res.status(405).json([]);
   }
 
-  const connection = req.body.connection as Connection;
+  const connection = changeTiDBConnectionToMySQL(req.body.connection as Connection);
```
Comment (boojack) [inline src/pages/api/connection/db.ts:14]: The logic seems a bit odd. Can we first check if the connection type is `tidb` and do the transform if it is?

Comment (Icemap) [reply src/pages/api/connection/db.ts:14]: Now I check it in the`changeTiDBConnectionToMySQL` function. I will move this logic out of this function.

File: src/utils/tidb.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { Connection } from "@/types";
+import { Engine } from "@/types/connection";
+
+type changeTiDBConnectionToMySQL = (connection: Connection) => Connection;
```
Comment (boojack) [inline src/utils/tidb.ts:4]: Looks like a unused type declare

**Review by boojack — APPROVED** [review]
LGTM, thanks for your contribution!

File: src/utils/tidb.ts (hunk 2 above)
Comment (Icemap) [reply src/utils/tidb.ts]: Deleted. Good catch!

**Pull request merged**